### ISSUE TITLE
Incorrect refactoring - Sourcery assumes attributes passed into calls are never functions

### ISSUE THREAD

**Issue by Sklavit** [sourcery-ai/sourcery#64]
### Issue description or question

The code was proposed to refactor
```
old_x = a.x
a.x = y
result = fund(a, **kwargs)
a.x = old_x
return result
```
proposed refactoring "Inline variable that is immediately returned" is incorrect:
```
old_x = a.x
a.x = y
a.x = old_x
return fund(a, **kwargs)
```

### Sourcery Version
0.7.4

Often available in Sourcery settings

### Code editor or IDE name and version

Pycharm v2020.1.2 pro

### OS name and version

Linux 18.04

**Comment by brendanator** [comment]
Thanks for the report, this is definitely a bug

**Comment by brendanator** [comment]
I've tried reproducing this and I'm not getting suggestions. Is this the exact code that is proposing for you or is it something slightly different?

**Comment by Sklavit** [comment]
![image](https://user-images.githubusercontent.com/1873961/87776279-2f8d5b80-c830-11ea-971a-cb738e7a83ac.png)

**Comment by Sklavit** [comment]
Oh, I see the difference: I pass in function not `client` but `client.patch` (which is a method).

**Comment by brendanator** [comment]
Oh I see. So `client.patch` is a method which is called and updates `client.api_token`, and therefore those 2 lines cannot be swapped.

This is a really difficult one for us in terms of the analysis. 
- In the original example you posted we didn't propose because `fund` is passed `a` and we assume that it might set `a.x`. 
- In the second example we assume that `client.patch` is unrelated to `client.api_token`, which is true almost all of the time. Unfortunately in this case where `client.patch` is a method that has access to its own `self` and via that to `client.api_token`

I'll have a think to see how to fix this one. Definitely a tricky one

Until then as a work around you can add a comment to the top of the method:
```python
# sourcery skip: inline-immediately-returned-variable
```

**Comment by brendanator** [comment]
Have found a fix for this. It'll be in the next release. Thanks for the report!

**Comment by Hellebore** [comment]
resolved with release of 0.7.9
